src/ape/managers/project.py

```diff
@@ -1,34 +1,78 @@
+import json
+from hashlib import md5
 from pathlib import Path
-from typing import Dict, List
+from typing import Dict, List, Optional
 
+import requests
 from dataclassy import dataclass
 
-from ape.types import ContractType  # Compiler, PackageManifest, PackageMeta
+from ape.types import Compiler, ContractType, PackageManifest, Source, Checksum # PackageMeta
 
 from .compilers import CompilerManager
 from .config import ConfigManager
 
 
+def compute_checksum(source: str, algorithm: str = "md5") -> str:
+    if algorithm == "md5":
+        hasher = md5()
+        hasher.update(source.encode("utf-8"))
+        return hasher.hexdigest()
+
+    else:
+        raise  # Unknown algorithm
+
 @dataclass
 class ProjectManager:
     path: Path
     config: ConfigManager
     compilers: CompilerManager
 
-    depedendencies: Dict[str, "ProjectManager"] = dict()
+    dependencies: Dict[str, "ProjectManager"] = dict()
 
     def __init__(self):
-        pass  # Look for depedencies from config
+        if isinstance(self.path, str):
+            # TODO: Handle URIs for external package references
+            self.path = Path(self.path)
+
+        self.dependencies = {
+            manifest.name: manifest
+            for manifest in map(self._extract_manifest, self.config.dependencies)
+        }
+
+    def _extract_manifest(self, manifest_uri: str) -> PackageManifest:
```

Comment:
Technically, dependencies is a dict of `str: ProjectManager`, that might need to get updated

Reply:
Updated the type to be `Dict[str, PackageManifest]`.